pkg/drain/nodegroup.go

```diff
@@ -127,35 +136,58 @@ func (n *NodeGroupDrainer) Drain() error {
 			}
 
 			if newPendingNodes.Len() == 0 {
-				logger.Success("drained all nodes: %v", drainedNodes.List())
+				waitForAllRoutinesToFinish(ctx, sem, parallelLimit)
+				logger.Success("drained all nodes: %v", mapToList(drainedNodes.Items()))
 				return nil // no new nodes were seen
 			}
 
-			logger.Debug("already drained: %v", drainedNodes.List())
+			logger.Debug("already drained: %v", mapToList(drainedNodes.Items()))
 			logger.Debug("will drain: %v", newPendingNodes.List())
-
 			for i, node := range newPendingNodes.List() {
-				pending, err := n.evictPods(node)
-				if err != nil {
-					logger.Warning("pod eviction error (%q) on node %s", err, node)
-					time.Sleep(retryDelay)
-					continue
-				}
-				logger.Debug("%d pods to be evicted from %s", pending, node)
-				if pending == 0 {
-					drainedNodes.Insert(node)
+				if err := sem.Acquire(ctx, 1); err != nil {
```

Comment:
There's a slight inconsistency here with which `context` to use - this one uses the one with `timeout` and L121 uses `context.TODO()`. Based on your comment `//need to use a different context` I'm not sure whether you meant to use a `context.TODO()` in both cases or to change that to something else? 🤔 Maybe `sem` prefers the one with the timeout?

Reply:
Uh, sorry for answering this one Jake. :D Feel free to delete me. :D